complex/ft_main.c

```diff
@@ -328,13 +328,19 @@ static int ft_fw_process_list(struct fi_info *hints, struct fi_info *info)
 	for (subindex = 1, fabric_info = info; fabric_info;
 	     fabric_info = fabric_info->next, subindex++) {
 
-		printf("Starting test %d-%d: ", series->test_index, subindex);
-		ft_show_test_info();
+		//provider name comparison needed to skip utility providers
+		if (strcmp(hints->fabric_attr->prov_name,
+		    fabric_info->fabric_attr->prov_name))
+			continue;
```

Comment:
Thanks for the changes. One more comment: I think the string comparison check might not be enough. If the user wishes to run ubertest for an utility provider, the corresponding fi_info will always be skipped since libfabric is going to  prefix the prov_name with the core provider name.

e.g.
user requests hints->fabric_attr->prov_name = "ofi-rxm"
libfabric returns fabric_info->fabric_attr->prov_name = "sockets;ofi-rxm"
and the string comparison check would fail.
We would need to check if the user has requested an utility provider and if so we want to do strcmp just on the util portion of returned prov_name. The ft_core_name and ft_util_name functions can be used for this.